sdk/messaging/azservicebus/admin/admin_client.go

```diff
@@ -214,3 +215,22 @@ func (ep *entityPager[TFeed, T, TOutput]) Fetcher(ctx context.Context) ([]TOutpu
 
 	return finalItems, nil
 }
+
+// mapATOMError checks if the error is a legitimate 404 or a "fake" 404 (where the service succeeded but gave us back an
+// empty feed instead). This "fake" behavior comes about because the API here is not truly a CRUD API (it's extremely close)
+// so we have to do some small workarounds.
+// NOTE: we had a debate about whether to return a nil instance or try to fabricate an HTTP 404 response instead (even if
+// one didn't come back) and went with 'nil' to avoid having a fake HTTP response, which would have been confusing.
+func mapATOMError[T any](err error) (*T, error) {
```

Comment:
Can this type parameter be removed?  The `T` isn't actually used, it's always `nil`.

Reply:
I want to leave it since it makes this code more compact:

https://github.com/Azure/azure-sdk-for-go/pull/17616/files/7642f0d8b6bdb6cbf41e726014e5533c9c9606bc#diff-c0a58bbe8d6af602f609af18f19b0873cb94974f9ae461a8a12ab800cb8111f3R183

```go
func (ac *Client) GetQueue(ctx context.Context, queueName string, options *GetQueueOptions) (*GetQueueResponse, error) {
  // other code
	if err != nil {
		return mapATOMError[GetQueueResponse](err)
  }
}
```

Otherwise I'd need to do another check to return the right type of nil, etc...